.github/workflows/buildAndTest.yaml

```diff
@@ -0,0 +1,24 @@
+name: Build And Test
+
+on: [push, pull_request]
+
+jobs:
+  buildAndTest:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - uses: actions/cache@v1
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Build
+      run: ./mvnw compile
+
+    - name: Unit Tests
+      run: ./mvnw test --fail-at-end
```

Comment:
This line would be:
```
  ./mvnw test -B -Dit.pubsub-emulator
```
I'm not sure if we want `--fail-at-end`. I guess it will run all the tests despite intermediate failures.

Reply:
I discovered `--fail-at-end` which it halted on first failure. I definitely think it's useful to run the full suite for a PR (that way we know if multiple are broken at once).

As for the pubsub emulator, that feels like more of an integration test to me. My plan is to expand these steps out farther, but hopefully in a way we can encapsulate them (say, with an `integration-test` target or something to separate them from unit tests).